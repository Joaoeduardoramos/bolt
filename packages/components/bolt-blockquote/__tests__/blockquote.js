import {
  isConnected,
  render,
  renderString,
  stopServer,
  html,
  vrtDefaultConfig as vrtConfig,
} from '../../../testing/testing-helpers';
const { readYamlFileSync } = require('@bolt/build-tools/utils/yaml');
const { join } = require('path');
const schema = readYamlFileSync(join(__dirname, '../blockquote.schema.yml'));
const { tag } = schema.properties;

const vrtDefaultConfig = Object.assign(vrtConfig, {
  failureThreshold: '0.01',
});

const timeout = 90000;

async function clipScreenshot(boundingBox, page) {
  return {
    clip: {
      x: boundingBox.x,
      y: boundingBox.y,
      width: Math.min(boundingBox.width, await page.viewport().width),
      height: Math.min(boundingBox.height, await page.viewport().height),
    },
  };
}

describe('button', () => {
  let page, context;

  afterAll(async () => {
    await stopServer();
  }, 100);

  beforeAll(async () => {
    context = await global.__BROWSER__.createIncognitoBrowserContext();
  });

  beforeEach(async () => {
    page = await context.newPage();
    await page.goto('http://127.0.0.1:4444/', {
      timeout: 0,
      waitLoad: true,
      waitNetworkIdle: true, // defaults to false
    });
  }, timeout);

  test('Basic usage', async () => {
    const results = await render(
      '@bolt-components-blockquote/blockquote.twig',
      {
        author: {
          name: 'Michelangelo di Lodovico Buonarroti Simoni',
          title: 'Renaissance Artist',
        },
        content:
          '<p>The greater danger for most of us lies not in setting our aim too high and falling short; but in setting our aim too low, and achieving our mark.</p>',
      },
      true,
    );
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('Default <bolt-blockquote> with Shadow DOM renders', async function() {
    const defaultBlockquoteOuter = await page.evaluate(() => {
      const blockquote = document.createElement('bolt-blockquote');
      blockquote.setAttribute('author-name', 'Chris Heilmann');
      blockquote.setAttribute('author-title', 'Developer Evangelist');
      blockquote.innerHTML = `
        <p>Java is to JavaScript what Car is to Carpet.</p>
      `;
      document.getElementById('root').appendChild(blockquote);
      document.getElementById('root').classList.add('u-bolt-padding-medium');
      blockquote.updated();
      return blockquote.outerHTML;
    });

    await page.evaluate(async () => {
      const selectors = Array.from(
        document.querySelectorAll('bolt-blockquote'),
      );
      return await Promise.all(
        selectors.map(blockquote => {
          if (blockquote._wasInitiallyRendered === true) return;
          return new Promise((resolve, reject) => {
            blockquote.addEventListener('ready', resolve);
            blockquote.addEventListener('error', reject);
          });
        }),
      );
    });

    const renderedHTML = await html(defaultBlockquoteOuter);

    const root = await page.$('#root');
    const image = await page.screenshot(
      await clipScreenshot(await root.boundingBox(), page),
    );

    expect(image).toMatchImageSnapshot(vrtDefaultConfig);

    expect(renderedHTML).toMatchSnapshot();
  });

  test('Default <bolt-blockquote> w/o Shadow DOM renders', async function() {
    const renderedBlockquoteHTML = await page.evaluate(() => {
      const blockquote = document.createElement('bolt-blockquote');
      blockquote.setAttribute(
        'author-name',
        'Michelangelo di Lodovico Buonarroti Simoni',
      );
      blockquote.setAttribute('align-items', 'center');
      blockquote.setAttribute('author-title', 'Renaissance Artist');
      blockquote.setAttribute('author-image', '/fixtures/500x500.jpg');
      blockquote.innerHTML = `<img slot="logo" src="/fixtures/paypal-logo.svg" alt="PayPal Logo">
        <p>Bolt Blockquote w/ Shadow DOM Manually Disabled</p>
        <p>Press any key to continue or any other key to quit.</p>
      `;
      document.getElementById('root').appendChild(blockquote);
      document.getElementById('root').classList.add('u-bolt-padding-medium');
      blockquote.useShadow = false;
      blockquote.updated();
      return blockquote.outerHTML;
    });

    const renderedHTML = await html(renderedBlockquoteHTML);

    expect(
      renderedHTML
        .querySelector('.c-bolt-blockquote')
        .classList.contains('c-bolt-blockquote--xlarge'),
    ).toBe(true);

    await page.evaluate(async () => {
      const selectors = Array.from(
        document.querySelectorAll('bolt-blockquote'),
      );
      return await Promise.all(
        selectors.map(blockquote => {
          if (blockquote._wasInitiallyRendered === true) return;
          return new Promise((resolve, reject) => {
            blockquote.addEventListener('ready', resolve);
            blockquote.addEventListener('error', reject);
          });
        }),
      );
    });

    const root = await page.$('#root');
    const image = await page.screenshot(
      await clipScreenshot(await root.boundingBox(), page),
    );

    expect(image).toMatchImageSnapshot(vrtDefaultConfig);

    expect(renderedHTML).toMatchSnapshot();
  });

  test('Default <bolt-blockquote> inside a <form> renders', async function() {
    const renderedBlockquoteHTML = await page.evaluate(() => {
      const form = document.createElement('form');
      const blockquote = document.createElement('bolt-blockquote');
      blockquote.setAttribute('author-name', 'Every Developer, Ever.');
      blockquote.setAttribute('border', 'horizontal');
      blockquote.setAttribute('size', 'xxlarge');
      blockquote.setAttribute('author-title', 'Web Developer');
      blockquote.innerHTML = `
        <img slot="logo" src="/fixtures/bolt-logo.svg" alt="Bolt Design System Logo" style="max-width: 200px">
        <p>Bolt Blockquote w/ Shadow DOM auto-disabled</p>
        <p>Yeah, but it works on my machine...</p>
      `;
      document.getElementById('root').appendChild(form);
      document.getElementById('root').classList.add('u-bolt-padding-medium');
      form.appendChild(blockquote);
      blockquote.updated();
      return blockquote.innerHTML;
    });

    await page.evaluate(async () => {
      const selectors = Array.from(
        document.querySelectorAll('bolt-blockquote'),
      );
      return await Promise.all(
        selectors.map(blockquote => {
          if (blockquote._wasInitiallyRendered === true) return;
          return new Promise((resolve, reject) => {
            blockquote.addEventListener('ready', resolve);
            blockquote.addEventListener('error', reject);
          });
        }),
      );
    });

    const renderedHTML = await html(renderedBlockquoteHTML);
    const root = await page.$('#root');
    const image = await page.screenshot(
      await clipScreenshot(await root.boundingBox(), page),
    );

    expect(image).toMatchImageSnapshot(vrtDefaultConfig);

    expect(renderedHTML).toMatchSnapshot();
  });
});
