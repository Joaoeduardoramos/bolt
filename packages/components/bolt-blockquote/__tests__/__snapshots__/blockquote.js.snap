// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-blockquote> Component advanced usage 1`] = `

<bolt-blockquote border="horizontal">
  <blockquote class="c-bolt-blockquote c-bolt-blockquote--xlarge c-bolt-blockquote--align-items-start c-bolt-blockquote--bordered-horizontal"
              is="shadow-root"
  >
    <replace-with-children class="c-bolt-blockquote__logo">
      <bolt-logo slot="logo"
                 class="c-bolt-logo--invert"
      >
        <bolt-image slot="logo"
                    src="/fixtures/paypal.svg"
                    srcset="/fixtures/paypal.svg"
                    sizes="auto"
                    ratio="124/33"
                    no-lazy
        >
          <bolt-ratio ratio="124/33">
            <replace-with-children class="c-bolt-ratio"
                                   style="padding-bottom: calc((33 / 124) * 100%); --aspect-ratio: 3.7575757575758;"
            >
              <img class="c-bolt-image__image"
                   srcset="/fixtures/paypal.svg"
                   sizes="auto"
                   src="/fixtures/paypal.svg"
              >
            </replace-with-children>
          </bolt-ratio>
        </bolt-image>
      </bolt-logo>
    </replace-with-children>
    <replace-with-grandchildren class="c-bolt-blockquote__quote">
      <div class="c-bolt-text c-bolt-text--semibold c-bolt-text--normal c-bolt-text--xlarge">
        <p>
          The greater danger for most of us lies not in setting our aim too high and falling short; but in setting our aim too low, and achieving our mark.
        </p>
      </div>
    </replace-with-grandchildren>
    <replace-with-grandchildren>
      <footer class="c-bolt-blockquote__footer">
        <replace-with-grandchildren class="c-bolt-blockquote__footer-item">
          <div class="c-bolt-blockquote__image">
            <bolt-image slot="author-image"
                        src="/fixtures/author.jpg"
                        srcset="/fixtures/author-50.jpg 50w, /fixtures/author-100.jpg 100w, /fixtures/author-200.jpg 200w, /fixtures/author-320.jpg 320w, /fixtures/author-480.jpg 480w"
                        sizes="auto"
                        ratio="500/500"
                        no-lazy
                        style="background-color: hsl(233, 33%, 97%);"
            >
              <bolt-ratio ratio="1/1">
                <replace-with-children class="c-bolt-ratio"
                                       style="padding-bottom: calc((1 / 1) * 100%); --aspect-ratio: 1;"
                >
                  <img class="c-bolt-image__image-placeholder"
                       sizes="auto"
                       src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
                  >
                  <img class="c-bolt-image__image"
                       srcset="/fixtures/author-50.jpg 50w, /fixtures/author-100.jpg 100w, /fixtures/author-200.jpg 200w, /fixtures/author-320.jpg 320w, /fixtures/author-480.jpg 480w"
                       sizes="auto"
                       src="/fixtures/author.jpg"
                  >
                </replace-with-children>
              </bolt-ratio>
            </bolt-image>
          </div>
        </replace-with-grandchildren>
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-name"
                class="c-bolt-text c-bolt-text--bold c-bolt-text--normal c-bolt-text--xsmall"
          >
            Michelangelo di Lodovico Buonarroti Simoni
          </cite>
        </replace-with-children>
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-title"
                class="c-bolt-text c-bolt-text--regular c-bolt-text--normal c-bolt-text--xsmall"
          >
            Renaissance Artist
          </cite>
        </replace-with-children>
      </footer>
    </replace-with-grandchildren>
  </blockquote>
</bolt-blockquote>

`;

exports[`<bolt-blockquote> Component align variations: center 1`] = `

<bolt-blockquote align-items="center">
  <blockquote class="c-bolt-blockquote c-bolt-blockquote--xlarge c-bolt-blockquote--align-items-center c-bolt-blockquote--bordered-vertical"
              is="shadow-root"
  >
    <replace-with-grandchildren class="c-bolt-blockquote__quote">
      <div class="c-bolt-text c-bolt-text--semibold c-bolt-text--normal c-bolt-text--xlarge">
        <p>
          The greater danger for most of us lies not in setting our aim too high and falling short; but in setting our aim too low, and achieving our mark.
        </p>
      </div>
    </replace-with-grandchildren>
    <replace-with-grandchildren>
      <footer class="c-bolt-blockquote__footer">
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-name"
                class="c-bolt-text c-bolt-text--bold c-bolt-text--normal c-bolt-text--xsmall"
          >
            Michelangelo di Lodovico Buonarroti Simoni
          </cite>
        </replace-with-children>
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-title"
                class="c-bolt-text c-bolt-text--regular c-bolt-text--normal c-bolt-text--xsmall"
          >
            Renaissance Artist
          </cite>
        </replace-with-children>
      </footer>
    </replace-with-grandchildren>
  </blockquote>
</bolt-blockquote>

`;

exports[`<bolt-blockquote> Component align variations: left 1`] = `

<bolt-blockquote align-items="left">
  <blockquote class="c-bolt-blockquote c-bolt-blockquote--xlarge c-bolt-blockquote--align-items-start c-bolt-blockquote--bordered-vertical"
              is="shadow-root"
  >
    <replace-with-grandchildren class="c-bolt-blockquote__quote">
      <div class="c-bolt-text c-bolt-text--semibold c-bolt-text--normal c-bolt-text--xlarge">
        <p>
          The greater danger for most of us lies not in setting our aim too high and falling short; but in setting our aim too low, and achieving our mark.
        </p>
      </div>
    </replace-with-grandchildren>
    <replace-with-grandchildren>
      <footer class="c-bolt-blockquote__footer">
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-name"
                class="c-bolt-text c-bolt-text--bold c-bolt-text--normal c-bolt-text--xsmall"
          >
            Michelangelo di Lodovico Buonarroti Simoni
          </cite>
        </replace-with-children>
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-title"
                class="c-bolt-text c-bolt-text--regular c-bolt-text--normal c-bolt-text--xsmall"
          >
            Renaissance Artist
          </cite>
        </replace-with-children>
      </footer>
    </replace-with-grandchildren>
  </blockquote>
</bolt-blockquote>

`;

exports[`<bolt-blockquote> Component align variations: right 1`] = `

<bolt-blockquote align-items="right">
  <blockquote class="c-bolt-blockquote c-bolt-blockquote--xlarge c-bolt-blockquote--align-items-end c-bolt-blockquote--bordered-vertical"
              is="shadow-root"
  >
    <replace-with-grandchildren class="c-bolt-blockquote__quote">
      <div class="c-bolt-text c-bolt-text--semibold c-bolt-text--normal c-bolt-text--xlarge">
        <p>
          The greater danger for most of us lies not in setting our aim too high and falling short; but in setting our aim too low, and achieving our mark.
        </p>
      </div>
    </replace-with-grandchildren>
    <replace-with-grandchildren>
      <footer class="c-bolt-blockquote__footer">
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-name"
                class="c-bolt-text c-bolt-text--bold c-bolt-text--normal c-bolt-text--xsmall"
          >
            Michelangelo di Lodovico Buonarroti Simoni
          </cite>
        </replace-with-children>
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-title"
                class="c-bolt-text c-bolt-text--regular c-bolt-text--normal c-bolt-text--xsmall"
          >
            Renaissance Artist
          </cite>
        </replace-with-children>
      </footer>
    </replace-with-grandchildren>
  </blockquote>
</bolt-blockquote>

`;

exports[`<bolt-blockquote> Component basic usage 1`] = `

<bolt-blockquote>
  <blockquote class="c-bolt-blockquote c-bolt-blockquote--xlarge c-bolt-blockquote--align-items-start c-bolt-blockquote--bordered-vertical"
              is="shadow-root"
  >
    <replace-with-grandchildren class="c-bolt-blockquote__quote">
      <div class="c-bolt-text c-bolt-text--semibold c-bolt-text--normal c-bolt-text--xlarge">
        <p>
          The greater danger for most of us lies not in setting our aim too high and falling short; but in setting our aim too low, and achieving our mark.
        </p>
      </div>
    </replace-with-grandchildren>
    <replace-with-grandchildren>
      <footer class="c-bolt-blockquote__footer">
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-name"
                class="c-bolt-text c-bolt-text--bold c-bolt-text--normal c-bolt-text--xsmall"
          >
            Michelangelo di Lodovico Buonarroti Simoni
          </cite>
        </replace-with-children>
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-title"
                class="c-bolt-text c-bolt-text--regular c-bolt-text--normal c-bolt-text--xsmall"
          >
            Renaissance Artist
          </cite>
        </replace-with-children>
      </footer>
    </replace-with-grandchildren>
  </blockquote>
</bolt-blockquote>

`;

exports[`<bolt-blockquote> Component border variations with inset: horizontal 1`] = `

<bolt-blockquote border="horizontal">
  <blockquote class="c-bolt-blockquote c-bolt-blockquote--xlarge c-bolt-blockquote--align-items-start c-bolt-blockquote--bordered-horizontal"
              is="shadow-root"
  >
    <replace-with-grandchildren class="c-bolt-blockquote__quote">
      <div class="c-bolt-text c-bolt-text--semibold c-bolt-text--normal c-bolt-text--xlarge">
        <p>
          The greater danger for most of us lies not in setting our aim too high and falling short; but in setting our aim too low, and achieving our mark.
        </p>
      </div>
    </replace-with-grandchildren>
    <replace-with-grandchildren>
      <footer class="c-bolt-blockquote__footer">
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-name"
                class="c-bolt-text c-bolt-text--bold c-bolt-text--normal c-bolt-text--xsmall"
          >
            Michelangelo di Lodovico Buonarroti Simoni
          </cite>
        </replace-with-children>
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-title"
                class="c-bolt-text c-bolt-text--regular c-bolt-text--normal c-bolt-text--xsmall"
          >
            Renaissance Artist
          </cite>
        </replace-with-children>
      </footer>
    </replace-with-grandchildren>
  </blockquote>
</bolt-blockquote>

`;

exports[`<bolt-blockquote> Component border variations with inset: none 1`] = `

<bolt-blockquote border="none">
  <blockquote class="c-bolt-blockquote c-bolt-blockquote--xlarge c-bolt-blockquote--align-items-start c-bolt-blockquote--borderless"
              is="shadow-root"
  >
    <replace-with-grandchildren class="c-bolt-blockquote__quote">
      <div class="c-bolt-text c-bolt-text--semibold c-bolt-text--normal c-bolt-text--xlarge">
        <p>
          The greater danger for most of us lies not in setting our aim too high and falling short; but in setting our aim too low, and achieving our mark.
        </p>
      </div>
    </replace-with-grandchildren>
    <replace-with-grandchildren>
      <footer class="c-bolt-blockquote__footer">
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-name"
                class="c-bolt-text c-bolt-text--bold c-bolt-text--normal c-bolt-text--xsmall"
          >
            Michelangelo di Lodovico Buonarroti Simoni
          </cite>
        </replace-with-children>
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-title"
                class="c-bolt-text c-bolt-text--regular c-bolt-text--normal c-bolt-text--xsmall"
          >
            Renaissance Artist
          </cite>
        </replace-with-children>
      </footer>
    </replace-with-grandchildren>
  </blockquote>
</bolt-blockquote>

`;

exports[`<bolt-blockquote> Component border variations with inset: vertical 1`] = `

<bolt-blockquote border="vertical">
  <blockquote class="c-bolt-blockquote c-bolt-blockquote--xlarge c-bolt-blockquote--align-items-start c-bolt-blockquote--bordered-vertical"
              is="shadow-root"
  >
    <replace-with-grandchildren class="c-bolt-blockquote__quote">
      <div class="c-bolt-text c-bolt-text--semibold c-bolt-text--normal c-bolt-text--xlarge">
        <p>
          The greater danger for most of us lies not in setting our aim too high and falling short; but in setting our aim too low, and achieving our mark.
        </p>
      </div>
    </replace-with-grandchildren>
    <replace-with-grandchildren>
      <footer class="c-bolt-blockquote__footer">
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-name"
                class="c-bolt-text c-bolt-text--bold c-bolt-text--normal c-bolt-text--xsmall"
          >
            Michelangelo di Lodovico Buonarroti Simoni
          </cite>
        </replace-with-children>
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-title"
                class="c-bolt-text c-bolt-text--regular c-bolt-text--normal c-bolt-text--xsmall"
          >
            Renaissance Artist
          </cite>
        </replace-with-children>
      </footer>
    </replace-with-grandchildren>
  </blockquote>
</bolt-blockquote>

`;

exports[`<bolt-blockquote> Component border variations: horizontal 1`] = `

<bolt-blockquote border="horizontal">
  <blockquote class="c-bolt-blockquote c-bolt-blockquote--xlarge c-bolt-blockquote--align-items-start c-bolt-blockquote--bordered-horizontal"
              is="shadow-root"
  >
    <replace-with-grandchildren class="c-bolt-blockquote__quote">
      <div class="c-bolt-text c-bolt-text--semibold c-bolt-text--normal c-bolt-text--xlarge">
        <p>
          The greater danger for most of us lies not in setting our aim too high and falling short; but in setting our aim too low, and achieving our mark.
        </p>
      </div>
    </replace-with-grandchildren>
    <replace-with-grandchildren>
      <footer class="c-bolt-blockquote__footer">
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-name"
                class="c-bolt-text c-bolt-text--bold c-bolt-text--normal c-bolt-text--xsmall"
          >
            Michelangelo di Lodovico Buonarroti Simoni
          </cite>
        </replace-with-children>
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-title"
                class="c-bolt-text c-bolt-text--regular c-bolt-text--normal c-bolt-text--xsmall"
          >
            Renaissance Artist
          </cite>
        </replace-with-children>
      </footer>
    </replace-with-grandchildren>
  </blockquote>
</bolt-blockquote>

`;

exports[`<bolt-blockquote> Component border variations: none 1`] = `

<bolt-blockquote border="none">
  <blockquote class="c-bolt-blockquote c-bolt-blockquote--xlarge c-bolt-blockquote--align-items-start c-bolt-blockquote--borderless"
              is="shadow-root"
  >
    <replace-with-grandchildren class="c-bolt-blockquote__quote">
      <div class="c-bolt-text c-bolt-text--semibold c-bolt-text--normal c-bolt-text--xlarge">
        <p>
          The greater danger for most of us lies not in setting our aim too high and falling short; but in setting our aim too low, and achieving our mark.
        </p>
      </div>
    </replace-with-grandchildren>
    <replace-with-grandchildren>
      <footer class="c-bolt-blockquote__footer">
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-name"
                class="c-bolt-text c-bolt-text--bold c-bolt-text--normal c-bolt-text--xsmall"
          >
            Michelangelo di Lodovico Buonarroti Simoni
          </cite>
        </replace-with-children>
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-title"
                class="c-bolt-text c-bolt-text--regular c-bolt-text--normal c-bolt-text--xsmall"
          >
            Renaissance Artist
          </cite>
        </replace-with-children>
      </footer>
    </replace-with-grandchildren>
  </blockquote>
</bolt-blockquote>

`;

exports[`<bolt-blockquote> Component border variations: vertical 1`] = `

<bolt-blockquote border="vertical">
  <blockquote class="c-bolt-blockquote c-bolt-blockquote--xlarge c-bolt-blockquote--align-items-start c-bolt-blockquote--bordered-vertical"
              is="shadow-root"
  >
    <replace-with-grandchildren class="c-bolt-blockquote__quote">
      <div class="c-bolt-text c-bolt-text--semibold c-bolt-text--normal c-bolt-text--xlarge">
        <p>
          The greater danger for most of us lies not in setting our aim too high and falling short; but in setting our aim too low, and achieving our mark.
        </p>
      </div>
    </replace-with-grandchildren>
    <replace-with-grandchildren>
      <footer class="c-bolt-blockquote__footer">
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-name"
                class="c-bolt-text c-bolt-text--bold c-bolt-text--normal c-bolt-text--xsmall"
          >
            Michelangelo di Lodovico Buonarroti Simoni
          </cite>
        </replace-with-children>
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-title"
                class="c-bolt-text c-bolt-text--regular c-bolt-text--normal c-bolt-text--xsmall"
          >
            Renaissance Artist
          </cite>
        </replace-with-children>
      </footer>
    </replace-with-grandchildren>
  </blockquote>
</bolt-blockquote>

`;

exports[`<bolt-blockquote> Component indent usage 1`] = `

<bolt-blockquote indent>
  <blockquote class="c-bolt-blockquote c-bolt-blockquote--xlarge c-bolt-blockquote--align-items-start c-bolt-blockquote--bordered-vertical c-bolt-blockquote--indented"
              is="shadow-root"
  >
    <replace-with-grandchildren class="c-bolt-blockquote__quote">
      <div class="c-bolt-text c-bolt-text--semibold c-bolt-text--normal c-bolt-text--xlarge">
        <p>
          The greater danger for most of us lies not in setting our aim too high and falling short; but in setting our aim too low, and achieving our mark.
        </p>
      </div>
    </replace-with-grandchildren>
    <replace-with-grandchildren>
      <footer class="c-bolt-blockquote__footer">
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-name"
                class="c-bolt-text c-bolt-text--bold c-bolt-text--normal c-bolt-text--xsmall"
          >
            Michelangelo di Lodovico Buonarroti Simoni
          </cite>
        </replace-with-children>
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-title"
                class="c-bolt-text c-bolt-text--regular c-bolt-text--normal c-bolt-text--xsmall"
          >
            Renaissance Artist
          </cite>
        </replace-with-children>
      </footer>
    </replace-with-grandchildren>
  </blockquote>
</bolt-blockquote>

`;

exports[`<bolt-blockquote> Component size variations: large 1`] = `

<bolt-blockquote size="large">
  <blockquote class="c-bolt-blockquote c-bolt-blockquote--large c-bolt-blockquote--align-items-start c-bolt-blockquote--bordered-vertical"
              is="shadow-root"
  >
    <replace-with-grandchildren class="c-bolt-blockquote__quote">
      <div class="c-bolt-text c-bolt-text--semibold c-bolt-text--normal c-bolt-text--large">
        <p>
          The greater danger for most of us lies not in setting our aim too high and falling short; but in setting our aim too low, and achieving our mark.
        </p>
      </div>
    </replace-with-grandchildren>
    <replace-with-grandchildren>
      <footer class="c-bolt-blockquote__footer">
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-name"
                class="c-bolt-text c-bolt-text--bold c-bolt-text--normal c-bolt-text--xsmall"
          >
            Michelangelo di Lodovico Buonarroti Simoni
          </cite>
        </replace-with-children>
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-title"
                class="c-bolt-text c-bolt-text--regular c-bolt-text--normal c-bolt-text--xsmall"
          >
            Renaissance Artist
          </cite>
        </replace-with-children>
      </footer>
    </replace-with-grandchildren>
  </blockquote>
</bolt-blockquote>

`;

exports[`<bolt-blockquote> Component size variations: xlarge 1`] = `

<bolt-blockquote size="xlarge">
  <blockquote class="c-bolt-blockquote c-bolt-blockquote--xlarge c-bolt-blockquote--align-items-start c-bolt-blockquote--bordered-vertical"
              is="shadow-root"
  >
    <replace-with-grandchildren class="c-bolt-blockquote__quote">
      <div class="c-bolt-text c-bolt-text--semibold c-bolt-text--normal c-bolt-text--xlarge">
        <p>
          The greater danger for most of us lies not in setting our aim too high and falling short; but in setting our aim too low, and achieving our mark.
        </p>
      </div>
    </replace-with-grandchildren>
    <replace-with-grandchildren>
      <footer class="c-bolt-blockquote__footer">
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-name"
                class="c-bolt-text c-bolt-text--bold c-bolt-text--normal c-bolt-text--xsmall"
          >
            Michelangelo di Lodovico Buonarroti Simoni
          </cite>
        </replace-with-children>
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-title"
                class="c-bolt-text c-bolt-text--regular c-bolt-text--normal c-bolt-text--xsmall"
          >
            Renaissance Artist
          </cite>
        </replace-with-children>
      </footer>
    </replace-with-grandchildren>
  </blockquote>
</bolt-blockquote>

`;

exports[`<bolt-blockquote> Component size variations: xxlarge 1`] = `

<bolt-blockquote size="xxlarge">
  <blockquote class="c-bolt-blockquote c-bolt-blockquote--xxlarge c-bolt-blockquote--align-items-start c-bolt-blockquote--bordered-vertical"
              is="shadow-root"
  >
    <replace-with-grandchildren class="c-bolt-blockquote__quote">
      <div class="c-bolt-text c-bolt-text--semibold c-bolt-text--normal c-bolt-text--xxlarge">
        <p>
          The greater danger for most of us lies not in setting our aim too high and falling short; but in setting our aim too low, and achieving our mark.
        </p>
      </div>
    </replace-with-grandchildren>
    <replace-with-grandchildren>
      <footer class="c-bolt-blockquote__footer">
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-name"
                class="c-bolt-text c-bolt-text--bold c-bolt-text--normal c-bolt-text--xsmall"
          >
            Michelangelo di Lodovico Buonarroti Simoni
          </cite>
        </replace-with-children>
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-title"
                class="c-bolt-text c-bolt-text--regular c-bolt-text--normal c-bolt-text--xsmall"
          >
            Renaissance Artist
          </cite>
        </replace-with-children>
      </footer>
    </replace-with-grandchildren>
  </blockquote>
</bolt-blockquote>

`;
